sechub-commons-security-spring/src/test/java/com/mercedesbenz/sechub/spring/security/OAuth2TokenExpirationCalculatorTest.java

```diff
@@ -0,0 +1,176 @@
+                .isInstanceOf(NullPointerException.class).hasMessageContaining("defaultDuration");
+    }
+
+    @Test
+    void oAuth2AccessToken_null_throws_exception() {
+        /* prepare */
+        Instant now = mock();
+        Duration defaultDuration = mock();
+        Duration minimumTokenValidity = mock();
+
+        /* test */
+        assertThatThrownBy(() -> calculatorToTest.calculateAccessTokenDuration(now, defaultDuration, null, minimumTokenValidity))
+                .isInstanceOf(NullPointerException.class).hasMessageContaining("oAuth2AccessToken");
+    }
+
+    @Test
+    void minimumTokenValidity_null_throws_exception() {
+        /* prepare */
+        Instant now = mock();
+        Duration defaultDuration = mock();
+        OAuth2AccessToken oAuth2AccessToken = mock();
+
+        /* test */
+        assertThatThrownBy(() -> calculatorToTest.calculateAccessTokenDuration(now, defaultDuration, oAuth2AccessToken, null))
+                .isInstanceOf(NullPointerException.class).hasMessageContaining("minimumTokenValidity");
+    }
+
+    @Test
+    void minimum_token_validity_when_default_expiry_is_less_than_minimum_and_expires_at_is_null() {
+        /* prepare */
+        Instant now = Instant.now();
+        Duration minimumTokenValidity = DEFAULT_EXPIRY.plusDays(1);
+        Instant expectedMinimumTokenValidity = now.plusSeconds(minimumTokenValidity.getSeconds());
+        OAuth2AccessToken oAuth2AccessToken = mock();
+        when(oAuth2AccessToken.getExpiresAt()).thenReturn(null);
+
+        /* execute */
+        Instant calculatedDuration = calculatorToTest.calculateAccessTokenDuration(now, DEFAULT_EXPIRY, oAuth2AccessToken, minimumTokenValidity);
+
+        /* test */
+        assertThat(calculatedDuration).isEqualTo(expectedMinimumTokenValidity);
+    }
+
+    @Test
+    void on_authentication_success_assumes_default_expiry_when_default_is_greater_than_minimum_and_expires_at_is_null() {
+        /* prepare */
+        Instant now = Instant.now();
+        Duration minimumTokenValidity = DEFAULT_EXPIRY.minusMinutes(20);
+        Instant expectedDefaultExpiry = now.plusSeconds(DEFAULT_EXPIRY.getSeconds());
+        OAuth2AccessToken oAuth2AccessToken = mock();
+        when(oAuth2AccessToken.getExpiresAt()).thenReturn(null);
+
+        /* execute */
+        Instant calculatedDuration = calculatorToTest.calculateAccessTokenDuration(now, DEFAULT_EXPIRY, oAuth2AccessToken, minimumTokenValidity);
+
+        /* test */
+        assertThat(calculatedDuration).isEqualTo(expectedDefaultExpiry);
+    }
+
+    @Test
+    void on_authentication_success_assumes_expires_from_token_when_greater_than_default_and_minimum() {
+        /* prepare */
+        Instant now = Instant.now();
+        Duration minimumTokenValidity = DEFAULT_EXPIRY.plusDays(1);
+
+        Instant expiresAtFromToken = Instant.now().plusSeconds(minimumTokenValidity.plusDays(5).getSeconds());
```

Comment:
I would always use the existing `now` variable for later calculations instead of creating a Instant.now() for others. In former test you did it this way.

But just a suggestion.